src/main/kotlin/no/fdk/fdk_event_harvester/harvester/HarvesterActivity.kt

```diff
@@ -0,0 +1,63 @@
+package no.fdk.fdk_event_harvester.harvester
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.launch
+import no.fdk.fdk_event_harvester.adapter.HarvestAdminAdapter
+import no.fdk.fdk_event_harvester.rabbit.RabbitMQPublisher
+import no.fdk.fdk_event_harvester.service.UpdateService
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.util.*
+import javax.annotation.PostConstruct
+
+private val LOGGER = LoggerFactory.getLogger(HarvesterActivity::class.java)
+private const val HARVEST_ALL_ID = "all"
+
+@Service
+class HarvesterActivity(
+    private val harvestAdminAdapter: HarvestAdminAdapter,
+    private val harvester: EventHarvester,
+    private val publisher: RabbitMQPublisher,
+    private val updateService: UpdateService
+): CoroutineScope by CoroutineScope(Dispatchers.Default) {
+
+    @PostConstruct
```

Comment:
Kjør på med application event i stedet - se eksempel i fdk-sparql-service

Reply:
Prøvde å bruke application events når jeg satte opp dataservice-harvester, husker ikke helt hva problemet var, men var noe knot. En spesiell grunn til at vi ikke burde bruke PostConstruct?

Reply:
PostConstruct kjøres etter at bean er lagd, men før at applikasjonen er startet. Tenker bare at det er greit å kjøre ting når alt er startet opp. Se https://github.com/Informasjonsforvaltning/fdk-sparql-service/blob/db45507d5cd9c619242224075041a31ef8f7be18/src/main/java/no/fdk/service/FusekiService.java#L32 for en like enkel implementasjon

Reply:
Det jeg hadde problem med i fjor er sikkert fikset, gjør et forsøk på å endre til EventListener